src/main.ts

```diff
@@ -170,6 +175,91 @@ export async function latest_version(version_prefix: string): Promise<string> {
   })
 }
 
+function glob_to_regex(glob: string): RegExp {
+  return new RegExp(
+    glob.replace(/\./g, '\\.').replace(/\*/g, '.*').replace(/\?/g, '.')
+  )
+}
+
+function list_files_in_dir(directory: string, files: string[] = []): string[] {
+  fs.readdirSync(directory).forEach(file => {
+    const p = path.join(directory, file)
+    if (fs.statSync(p).isDirectory()) {
+      return list_files_in_dir(p, files)
+    } else {
+      return files.push(p)
+    }
+  })
+
+  return files
+}
+
+function dpkg_read_config(): DpkgConfig {
+  const dpkgConfigFilepath = '/etc/dpkg/dpkg.cfg'
+  const dpkgConfigDirPath = '/etc/dpkg/dpkg.cfg.d'
+
+  const configs = list_files_in_dir(dpkgConfigDirPath, [dpkgConfigFilepath])
+
+  const dpkgConfig: DpkgConfig = {
+    excludes: [],
+    includes: []
+  }
+
+  configs.forEach(config => {
+    try {
+      const dpkgExcludesFile = fs.readFileSync(config, 'utf8')
+      dpkgExcludesFile.split('\n').forEach(line => {
+        // Exclude
+        const excludePattern = 'path-exclude='
+        if (line.startsWith(excludePattern)) {
+          const regex = glob_to_regex(line.substring(excludePattern.length))
+          dpkgConfig.excludes.push(regex)
+          return
+        }
+
+        // Include
+        const includePattern = 'path-include='
+        if (line.startsWith(includePattern)) {
+          const regex = glob_to_regex(line.substring(includePattern.length))
+          dpkgConfig.includes.push(regex)
+          return
+        }
+      })
+    } catch (err) {
+      core.info(`dpkg excludes file not available: ${err}`)
+    }
+  })
+
+  return dpkgConfig
+}
+
+function dpkg_is_file_included(
+  dpkgConfig: DpkgConfig,
+  filepath: string
+): boolean {
+  var included = true
+
+  dpkgConfig.excludes.forEach(exclude => {
+    if (exclude.test(filepath)) {
+      included = false
+      return
+    }
+  })
+
+  if (included) {
+    return true
+  }
+
+  dpkgConfig.includes.forEach(include => {
+    if (include.test(filepath)) {
+      included = true
+      return
+    }
+  })
+
+  return included
+}
```

Comment:
AFAIU, we can't split `excludes` and `includes` into two arrays, because their order matters. From [man 1 dpkg](https://man7.org/linux/man-pages/man1/dpkg.1.html):

> --path-exclude=glob-pattern
> --path-include=glob-pattern
>
> \<...\>
>
> These two options can be specified multiple times, and
> interleaved with each other. Both are processed in the given
> order, with the last rule that matches a file name making the
> decision.

From [man 5 dpkg.cfg](https://man7.org/linux/man-pages/man5/dpkg.cfg.5.html):

> This file contains default options for dpkg. Each line contains a
> single option which is **exactly the same as a normal command line
> option** for dpkg except for the leading hyphens which are not used
> here.

In other words, includes have no precedence over excludes.

(TBH, I guess that your solution should anyway work for all the practical cases, but since we already have a general solution with parsing of the configuration and it is not a big deal to match the `dpkg` behavior exactly, it would be nice to implement it this way.)

----

While we're here, two typos from the commit message:

> This file is installed because it matches exclude-path in dpgk config,

* installed -> not installed
* dpgk -> dpkg

Reply:
It's not that easy:
consider the following case
```
--path-include=/usr/local/folder/file
--path-exclude=/usr/local/folder/*
```
In that order, should the `/usr/local/folder/file` be included or not? According to the man you provided - it should not, because only order is relevant (the more exact match doesn't win). There is also a problem if several config files with includes/excludes are present - the result is depended on the order in which the config files are processed. Shortly speaking - IMO their approach is bullshit) But I will try to replicate it as following:
1) Process config files in alphabetical order of the path
2) Create on array with structs like {type: 'exclude'/'include', path: 'regex'} in the order those options are read
3) For each file iterate over the array constructed in 2 with the final being the last found regex match and its type

Reply:
I've rewrote it according to the propose above see https://github.com/tarantool/setup-tarantool/compare/16f0e17a2a7fbaba50d742aaaafd5b00e4261f46..5a4e6640d41435e53d12468e113c834e0bead844

Reply:
I've no insight about precise algorithm: I know just what the man page says. Now it seems that our algorithm follows it.

Thank you for the update.